### PR TITLE
Regression Testing #2

### PR DESCRIPTION
**Pull request by kanye-quest**
This PR cleans up the commit history from #176.

Summary
======
- Adds CTest-based regression testing.
  - A test driver, `tests/glvis_driver.py`, runs glvis with a modified stream that generates screenshots.
    Two screenshots are generated: one with the original image from the stream, and one with the `*` key command applied to induce a slight change in the image.
    Screenshots can then be compared against a baseline with the structural similarity metric. A given test case passes if the original image is above the SSIM cutoff of 0.999 and the control image is below this cutoff.
  - Tests are run in headless mode for both `ubuntu-20.04` and `macos-10.15` in Github Actions, with streams and corresponding baselines available in [GLVis/data](https://github.com/GLVis/data)
- Bugfixes/miscellaneous changes:
  - Fix some issues with screenshots on macOS and Linux
  - Improve `SignalKeyDown()` handling for X11
  - Handle resizes synchronously in the main thread
  - Improve error handling with SDL

### CODE REVIEW

**Review by tzanio — APPROVED** [review]
Thanks so much @kanye-quest !

**Comment by tzanio** [comment]
Can you please resolve the conflicts (from https://github.com/GLVis/glvis/pull/190)?

File: tests/glvis_driver.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    if ret != 0:
+        print("[FAIL] GLVis exited with error code {}.".format(ret))
+        return False
+    if not os.path.exists(t_group):
+        os.mkdir(t_group)
+    output_name = "{0}/{1}.png".format(t_group, t_name)
+
+    ret = os.system("mv {0} {1}".format(screenshot_file, output_name))
+    if ret != 0:
+        print("[FAIL] Could not copy output image: exit code {}.".format(ret))
```
Comment (adrienbernede) [inline tests/glvis_driver.py:84]:
```suggestion
        print("[FAIL] Could not move output image: exit code {}.".format(ret))
```

**Review by adrienbernede — APPROVED** [review]
Looks great!
I tested it locally and it worked like a charm after settings things up (I have a python environment for glvis-testing).
I felt the need to insist on a bit on the developer workflow (make rebaseline) in the baseline details section.
Feel free to accept or reject.

**Comment by tzanio** [comment]
@kanye-quest, any ideas why the `builds / macos-10.15-optim-sequential-cmake` test is failing?

**Comment by tzanio** [comment]
@adrienbernede, can we change the GH actions names to be consistent with the new ones in mfem?

**Comment by tzanio** [comment]
I can't express how cool this new regression testing system is... thanks so much @kanye-quest !

**Pull request merged**